### PR TITLE
training.py _check_loss_and_target_compatibility() fix crash if y is None

### PR DESCRIPTION
**Pull request by ahundt**
Don't access `y` param of `model.fit()` if it is `None`. This prevents a crash when `model.fit(None,None, ...)` is called and loss is provided to `model.compile()`.

### CODE REVIEW

**Comment by ahundt** [comment]
I'm pretty sure this one is still valid

**Comment by ragulpr** [comment]
+1 for this pull, it seems necessary for various hacks/workarounds for target/predicted compatibility checks and in order to reuse inputs in loss (ex autoencoders w. embedding layers)

Great work! Hope it's merged

**Comment by ahundt** [comment]
@fchollet I know we passed on most of my other PRs once the target tensor parameter was added, but this one is trivially simple and worth merging.

**Comment by ahundt** [comment]
thanks!
